internal/store/interfaces.go

```diff
@@ -8,37 +8,11 @@ import (
 
 //go:generate mockgen -source ./interfaces.go -destination ./mocks/interfaces.go -package mocks StatusTrackingGateway,Gateway,RouteTrackingListener,Listener,StatusTrackingRoute,Route,Store
 
-// StatusTrackingGateway is an optional extension
-// of Gateway. If supported by a Store, when
-// a Gateway is synced to an external location,
-// its corresponding callbacks should
-// be called.
-type StatusTrackingGateway interface {
-	Gateway
-
-	TrackSync(ctx context.Context, sync func() (bool, error)) error
-}
-
 // Gateway describes a gateway.
 type Gateway interface {
 	ID() core.GatewayID
-	Bind(ctx context.Context, route Route) []string
-	Remove(ctx context.Context, id string) error
```

Comment:
These are now the responsibility of the `Binder` interface with `Bind`/`Unbind` methods. `Unbind` also incorporates the logic of the `OnGatewayRemoved` method below.